scene/gui/color_picker.cpp

```diff
@@ -1817,34 +1817,23 @@ void ColorPickerButton::_modal_closed() {
 void ColorPickerButton::pressed() {
 	_update_picker();
 
-	Size2 size = get_size() * get_viewport()->get_canvas_transform().get_scale();
+	Size2 minsize = popup->get_contents_minimum_size();
+	float viewport_height = get_viewport_rect().size.y;
 
 	popup->reset_size();
 	picker->_update_presets();
 	picker->_update_recent_presets();
 
-	Rect2i usable_rect = popup->get_usable_parent_rect();
-	//let's try different positions to see which one we can use
-
-	Rect2i cp_rect(Point2i(), popup->get_size());
-	for (int i = 0; i < 4; i++) {
-		if (i > 1) {
-			cp_rect.position.y = get_screen_position().y - cp_rect.size.y;
-		} else {
-			cp_rect.position.y = get_screen_position().y + size.height;
-		}
-
-		if (i & 1) {
-			cp_rect.position.x = get_screen_position().x;
-		} else {
-			cp_rect.position.x = get_screen_position().x - MAX(0, (cp_rect.size.x - size.x));
-		}
-
-		if (usable_rect.encloses(cp_rect)) {
-			break;
-		}
+	// Determine in which direction to show the popup. By default popup horizontally centered below the button.
+	// But if the popup doesn't fit below and the button is in the bottom half of the viewport, show above.
+	bool show_above = false;
+	if (get_global_position().y + get_size().y + minsize.y > viewport_height && get_global_position().y * 2 + get_size().y > viewport_height) {
```

Comment:
You are kind of mixing different coordinate systems in a single equation here.
- `get_global_position()`: Canvas coordinate system of the containing Canvas
- `get_size()`: Local coordinate system of the `ColorPickerButton`
- `minsize`: Viewport coordinate system of popup
- `viewport_height`: Viewport coordinates of the containing Viewport (actually I am not entirely sure about that one)

In the editor this is in many cases not a problem, but can you please test, if this still works, if the editor uses a non-default scale?

Reply:
Here is a MRP for stress-testing: [BugColorPicker.zip](https://github.com/godotengine/godot/files/10538275/BugColorPicker.zip)

I can have a look at the implementation regarding improvements, but unfortunately not right away.

Reply:
Aw, I kind of completely forgot this is a user-exposed thing. I'm honestly not sure what users would expect from it when they use it in their projects. But from a first glance at your test project, it doesn't look more broken than before

Reply:
So it could be a valid approach to fix the editor-issue now and take care of the user-exposed problems at a later time.